alternator/executor.cc

```diff
@@ -4548,8 +4548,10 @@ static future<std::vector<mutation>> create_keyspace(std::string_view keyspace_n
 }
 
 future<> executor::start() {
-    // Currently, nothing to do on initialization. We delay the keyspace
-    // creation (create_keyspace()) until a table is actually created.
+    exp_cache_stats_updater::init(&_stats,
+        [c = _expression_cache] () {return c->size();},
+        [c = _expression_cache] () {return c->memory_footprint();}
```

Comment:
Wow, loading_cache made this patch so ugly :-(
Instead of trivial stats being just an integers like in alternator/shats.hh, we have all this extra complexity and this line in particular too me three reads of this code to finally understand.

By the way, I wouldn't shed any tears if we didn't have stats for size and memory footprint. I think the hit/miss rate is important (you can see if something is bad if it's low), but what can I do about the memory footprint, if it's 1K or 100K, do we care? By the way, is the memory_footprint even correct? Who calculates it and how (given that the stored expressions are complex objects which do memory allocation, I doubt it is done correctly).

Reply:
I'll try to write new cache class (synchronous) in revisited implementation of this PR.

Reply:
Note, there's a reason to keep this asynchronous - if parsing is so slow that it can cause a stall. In this case we'd want to insert yields. But I expect it's not that slow (and we don't yield in the CQL parsing).

Reply:
for non-trivial expression with 10 fields it's 68k instructions, which is 20% of cpu for single node write path. But locally I haven't seen stalls reported, so it's not high enough I guess.

Reply:
Parsing shouldn't be very slow, and we even have a limit - 4096 bytes - on the length of the expression to parse (IIRC we don't have such a limit in CQL). I don't think we'll ever want to yield in the middle. But if you want to think ahead, maybe...

Reply:
I agree it's not fast (that's the whole raison d'etre for this PR ;-)), but the question is if it can ever take more than, SAY, half a millisecond. If it can't, then I think it's not interesting to yield in the middle.
Note that to yield inside ANTLR, we'll need to introduce a parsing seastar::thread - because it will be impossible to futurize ANTLR itself (as well as very inefficient in its highly recursive design). Doesn't sound very useful to me, but if you want to do it, feel free to go ahead.

Reply:
I think we can be synchronous for now. That alleviates most of the problems. We may go back to asynchronous one days, but make the entire path asynchronous, not just some random parts with abandoned futures in the middle.